pkg/controller.v1alpha3/experiment/suggestion/suggestion.go

```diff
@@ -1,31 +1,83 @@
 package suggestion
 
 import (
+	"context"
+	//"fmt"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	//utilrand "k8s.io/apimachinery/pkg/util/rand"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+
 	experimentsv1alpha3 "github.com/kubeflow/katib/pkg/apis/controller/experiments/v1alpha3"
-	api_pb "github.com/kubeflow/katib/pkg/apis/manager/v1alpha3"
-	common "github.com/kubeflow/katib/pkg/common/v1alpha3"
+	suggestionsv1alpha3 "github.com/kubeflow/katib/pkg/apis/controller/suggestions/v1alpha3"
 )
 
+var log = logf.Log.WithName("experiment-suggestion-client")
+
 type Suggestion interface {
-	GetSuggestions(instance *experimentsv1alpha3.Experiment, addCount int32) ([]*api_pb.Trial, error)
+	GetOrCreateSuggestion(instance *experimentsv1alpha3.Experiment, suggestionRequests int32) (*suggestionsv1alpha3.Suggestion, error)
+	UpdateSuggestion(suggestion *suggestionsv1alpha3.Suggestion, suggestionRequests int32) error
 }
 
 type General struct {
+	scheme *runtime.Scheme
+	client.Client
 }
 
-func New() Suggestion {
-	return &General{}
+func New(scheme *runtime.Scheme, client client.Client) Suggestion {
+	return &General{scheme: scheme, Client: client}
 }
 
-func (g *General) GetSuggestions(instance *experimentsv1alpha3.Experiment, addCount int32) ([]*api_pb.Trial, error) {
-	request := &api_pb.GetSuggestionsRequest{
-		ExperimentName: instance.Name,
-		AlgorithmName:  instance.Spec.Algorithm.AlgorithmName,
-		RequestNumber:  addCount,
-	}
-	if reply, err := common.GetSuggestions(request); err != nil {
+func (g *General) GetOrCreateSuggestion(instance *experimentsv1alpha3.Experiment, suggestionRequests int32) (*suggestionsv1alpha3.Suggestion, error) {
+	logger := log.WithValues("experiment", types.NamespacedName{Name: instance.Name, Namespace: instance.Namespace})
+	suggestion := &suggestionsv1alpha3.Suggestion{}
+	err := g.Get(context.TODO(),
+		types.NamespacedName{Name: instance.GetName(), Namespace: instance.GetNamespace()}, suggestion)
+	if err != nil && errors.IsNotFound(err) {
+		logger.Info("Creating Suggestion", "namespace", instance.Namespace, "name", instance.Name, "requests", suggestionRequests)
+		if err := g.createSuggestion(instance, suggestionRequests); err != nil {
+			return nil, err
```

Comment:
LGTM while I think we should have a comment to show why we should return an error here.

Reply:
Added logs